refactor(skills): replace arch-diagram-builder agent with reusable architecture-diagrams skill (#2068)
microsoft/hve-core: .github/skills/project-planning/adr-author/templates/diagram-ascii.md

Variant: ascii
Selected when: state.userPreferences.diagramFormat == "ascii"
Purpose: lightweight starting scaffold for box-and-arrow diagrams in ADRs

> IaC-derived diagrams: when the ADR diagram is derived from infrastructure source files (Terraform, Bicep, ARM), author it with the architecture-diagrams skill (.github/skills/hve-core/architecture-diagrams/SKILL.md) using ascii output and embed that result in place of this scaffold; that skill owns the ASCII conventions, arrow types, and boundary notation.

Reviewer: [portability] Hardcoded skill path in template metadata comment

This comment guides agents reading the template to load the skill at `.github/skills/hve-core/architecture-diagrams/SKILL.md`. That repo-root path does not exist in plugin or extension distributions. The new rule in `prompt-builder.instructions.md` (added by this PR) says to reference skills by name, not by hardcoded path.

```suggestion
IaC-derived diagrams: when the ADR diagram is derived from infrastructure source files (Terraform, Bicep, ARM), use the `architecture-diagrams` skill (load its SKILL.md) with ascii output and embed that result in place of this scaffold; that skill owns the ASCII conventions, arrow types, and boundary notation.
```

Author: Done. Removed the hardcoded `.github/skills/hve-core/architecture-diagrams/SKILL.md` path from the template metadata comment; it now references the `architecture-diagrams` skill by name so it stays portable across distributions. Pushed in 057e8313.
